fee-market/src/pairs.rs

```diff
@@ -0,0 +1,96 @@
+multiversx_sc::imports!();
+
+mod pair_proxy {
+    multiversx_sc::imports!();
+
+    #[multiversx_sc::proxy]
+    pub trait PairProxy {
+        #[view(getSafePriceByDefaultOffset)]
+        fn get_safe_price_by_default_offset(
+            &self,
+            pair_address: ManagedAddress,
+            input_payment: EsdtTokenPayment,
+        ) -> EsdtTokenPayment;
+    }
+}
+
+#[multiversx_sc::module]
+pub trait PairsModule {
+    #[only_owner]
+    #[endpoint(addPairs)]
+    fn add_pairs(
+        &self,
+        tokens_address_pairs: MultiValueEncoded<
+            MultiValue3<TokenIdentifier, TokenIdentifier, ManagedAddress>,
+        >,
+    ) {
+        for pairs in tokens_address_pairs {
```

Comment:
I think here we need to integrate an oracle contracts, as the sovereign cross chain SCs will not stay in shard 1 where the DEX is.

Reply:
we need a new interface and oracle contract which has all the prices. Maybe we could directly integrate here the oracle price contract, first update that, and then put it here.